### PR TITLE
fix(frontend): download detection audio by ID

### PR DESCRIPTION
**Pull request by ejhnsn**
## Summary

- fix the Detection Details “Download audio” link by using the detection ID endpoint instead of the filename endpoint
- preserve the browser download behavior and server-provided original filename/extension
- add a regression test that verifies the exact ID-based URL and download attribute

This uses the ID-based audio endpoint secured for Private Mode by #4083.

## Testing

- `golangci-lint run -v` (with the repository's CI build tags)
- `go test -race -short ./...` (with the repository's macOS CI build tags)
- `npm run check:all`
- `npm test -- --run` (181 files, 2,707 tests)

## Preflight Status

- [x] Reuse/efficiency review passed; reuses the established ID-based audio endpoint and `buildAppUrl`
- [x] Correctness and edge-case review passed
- [x] Code-quality review passed
- [x] i18n/accessibility review passed; existing translated label and aria label are preserved
- [x] Integration wiring review passed; the URL matches the detail-page player and backend route
- [x] Regression/backward-compatibility review passed; the filename endpoint remains available to existing clients
- [x] Full Go and frontend lint/test gates passed
- [x] Diff is limited to this single bug fix; no secrets, PII, TODOs, or breaking changes

## Summary by CodeRabbit

* **Improvements**
  * Updated audio downloads in detection details to use the modern ID-based API endpoint for more reliable access.

## Release notes
- audience: user
- summary: Fix the "Download audio" link on the detection details page so it reliably saves the clip, using the ID-based audio endpoint that resolves the correct file and original filename
- feature: none
- breaking: none
- config: none
- schema: none

### CODE REVIEW

File: frontend/src/lib/desktop/views/DetectionDetail.test.ts [hunk 1]
```diff
@@ -123,3 +123,43 @@ describe('DetectionDetail stale-response race (#978)', () => {
...
+    const { container } = detailTest.render({ detectionId: '1239' });
+
+    await waitFor(() => {
+      expect(container.querySelector('a.meta-download')).not.toBeNull();
+    });
+
+    const downloadLink = container.querySelector<HTMLAnchorElement>('a.meta-download');
+    expect(downloadLink?.getAttribute('href')).toBe('/api/v2/audio/1239');
+    expect(downloadLink).toHaveAttribute('download');
+  });
```
[the remark this answers is not in the document]
Comment (ejhnsn) [reply frontend/src/lib/desktop/views/DetectionDetail.test.ts:166]: Thanks for flagging the contract. The client intentionally does not propose a filename here: the bare `download` attribute renders as `download=""`, allowing the ID endpoint’s `Content-Disposition filename*` response to supply the canonical original name and actual extension. That server path is covered by `TestServeAudioByID` and `TestContentDispositionFilename`; it also removes the misleading legacy timestamp `Z` per #4030, so assigning the fixture’s raw `...Z.m4a` name client-side would be stale. I tightened the component assertion in def0684d2 to require the exact empty attribute and added a comment explaining the delegation.
